ocis-pkg/CHANGELOG.md

```diff
@@ -0,0 +1,318 @@
+
+
+* Change - Provide cache for roles: [#59](https://github.com/owncloud/ocis-pkg/pull/59)
+
+   In order to work efficiently with permissions we provide a cache for roles and a middleware to
+   update the cache based on roleIDs from the metadata context. It can be used to check permissions
+   in service handlers.
+
+   https://github.com/owncloud/ocis-pkg/pull/59
+
+
+* Change - Roles manager: [#60](https://github.com/owncloud/ocis-pkg/pull/60)
+
+   We combined the roles middleware and cache into a roles manager. The manager doesn't expose the
+   cache anymore and manages the state of the cache by fetching roles from the role service which
+   don't exist in the cache, yet.
+
+   https://github.com/owncloud/ocis-pkg/pull/60
+
+# Changelog for [2.4.0] (2020-08-25)
+
+The following sections list the changes in ocis-pkg 2.4.0.
+
+[2.4.0]: https://github.com/owncloud/ocis-pkg/compare/v2.3.0...v2.4.0
+
+## Summary
+
+* Change - Use go-micro's metadata context for account id: [#56](https://github.com/owncloud/ocis-pkg/pull/56)
+
+## Details
+
+* Change - Use go-micro's metadata context for account id: [#56](https://github.com/owncloud/ocis-pkg/pull/56)
+
+   We switched to using go-micro's metadata context for reliably passing the AccountID in the
+   context across service boundaries.
+
+   https://github.com/owncloud/ocis-pkg/pull/56
+
+# Changelog for [2.3.0] (2020-08-17)
+
+The following sections list the changes in ocis-pkg 2.3.0.
+
+[2.3.0]: https://github.com/owncloud/ocis-pkg/compare/v2.2.1...v2.3.0
+
+## Summary
+
+* Bugfix - Remove redigo 2.0.0+incompatible dependency: [#33](https://github.com/owncloud/ocis-graph/pull/33)
+* Change - Add middleware for x-access-token distmantling: [#46](https://github.com/owncloud/ocis-pkg/pull/46)
+* Enhancement - Add `ocis.id` and numeric id claims: [#50](https://github.com/owncloud/ocis-pkg/pull/50)
+
+## Details
+
+* Bugfix - Remove redigo 2.0.0+incompatible dependency: [#33](https://github.com/owncloud/ocis-graph/pull/33)
+
+   Get rid of redigo v2.0.0
+
+   https://github.com/owncloud/ocis-graph/pull/33
+
+
+* Change - Add middleware for x-access-token distmantling: [#46](https://github.com/owncloud/ocis-pkg/pull/46)
+
+   We added a middleware that dismantles the `x-access-token` from the request header and makes
+   it available in the context.
+
+   https://github.com/owncloud/ocis-pkg/pull/46
+
+
+* Enhancement - Add `ocis.id` and numeric id claims: [#50](https://github.com/owncloud/ocis-pkg/pull/50)
+
+   We added an `ocis.id` claim to the OIDC standard claims. It allows the idp to send a stable
+   identifier that can be exposed to the outside world (in contrast to sub, which might change
+   whens the IdP changes).
+
+   In addition we added `uidnumber` and `gidnumber` claims, which can be used by the IdP as well.
+   They will be used by storage providers that integrate with an existing LDAP server.
+
+   https://github.com/owncloud/ocis-pkg/pull/50
+
+# Changelog for [2.2.1] (2020-04-21)
```

Comment:
Codacy found an issue: [[no-shortcut-reference-link] Use the trailing `[]` on reference links](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6202733)